spec/datadog/integration_spec.rb

```diff
@@ -23,22 +23,30 @@ def wait_for_tracer_sent
 
     context 'for threads' do
       let(:original_threads) { Thread.list }
-      let!(:original_threads_inspect) { inspect_threads(original_threads) } # Store result as stack trace will change
-      let(:threads) { Thread.list }
+      let(:start_tracer) do
+        Datadog::Tracing.trace('test.op') {}
+
+        new_threads = Thread.list - original_threads
+        tracer_threads.concat(new_threads.map(&:object_id))
```

Comment:
I'm not quite sure I follow the need to use `object_id`s instead of the thread objects. What's the intention?

If you're worried that keeping the actual thread objects referenced is a problem, it's actually not: `Thread.list` only returns threads that are alive, even though more `Thread` objects may still exist:

```ruby
$ pry
[1] pry(main)> Thread.list
=> [#<Thread:0x0000786c3b447c60 run>]
[2] pry(main)> thread = Thread.new { sleep 2 }
=> #<Thread:0x0000786c3b195b20 (pry):2 run>
[3] pry(main)> Thread.list
=> [#<Thread:0x0000786c3b447c60 run>, #<Thread:0x0000786c3b195b20 (pry):2 sleep>]
[4] pry(main)> sleep 2
=> 2
[5] pry(main)> Thread.list
=> [#<Thread:0x0000786c3b447c60 run>]
[6] pry(main)> thread
=> #<Thread:0x0000786c3b195b20 (pry):2 dead>
```

Reply:
By default, I prefer to compare identifiers instead of objects, but you're correct that it's unnecessary here.

Reply:
Tip: You don't need to manually compare `object_id`, RSpec has helpers for that: https://dev.to/andy4thehuynh/the-difference-between-rspec-eq-and-be-3jep :)